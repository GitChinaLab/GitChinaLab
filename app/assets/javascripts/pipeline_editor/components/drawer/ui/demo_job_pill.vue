<script>
export default {
  props: {
    jobName: {
      type: String,
      required: true,
    },
  },
};
</script>
<template>
  <div
    class="gl-w-13 gl-h-6 gl-font-sm gl-bg-white gl-inset-border-1-blue-500 gl-text-center gl-text-truncate gl-rounded-pill gl-px-4 gl-py-2 gl-relative gl-z-index-1 gl-transition-duration-slow gl-transition-timing-function-ease"
  >
    {{ jobName }}
  </div>
</template>
