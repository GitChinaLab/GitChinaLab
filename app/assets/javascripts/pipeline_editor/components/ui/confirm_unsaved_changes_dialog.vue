<script>
export default {
  props: {
    hasUnsavedChanges: {
      type: Boolean,
      required: true,
    },
  },
  created() {
    window.addEventListener('beforeunload', this.confirmChanges);
  },
  destroyed() {
    window.removeEventListener('beforeunload', this.confirmChanges);
  },
  methods: {
    confirmChanges(e = {}) {
      if (this.hasUnsavedChanges) {
        e.preventDefault();
        // eslint-disable-next-line no-param-reassign
        e.returnValue = ''; // Chrome requires returnValue to be set
      }
    },
  },
  render: () => null,
};
</script>
