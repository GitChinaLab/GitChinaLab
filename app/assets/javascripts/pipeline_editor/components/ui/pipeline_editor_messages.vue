<script>
import { GlAlert } from '@gitlab/ui';
import { getParameterValues, removeParams } from '~/lib/utils/url_utility';
import { __, s__ } from '~/locale';
import {
  COMMIT_FAILURE,
  COMMIT_SUCCESS,
  DEFAULT_FAILURE,
  DEFAULT_SUCCESS,
  LOAD_FAILURE_UNKNOWN,
  PIPELINE_FAILURE,
} from '../../constants';
import CodeSnippetAlert from '../code_snippet_alert/code_snippet_alert.vue';
import {
  CODE_SNIPPET_SOURCE_URL_PARAM,
  CODE_SNIPPET_SOURCES,
} from '../code_snippet_alert/constants';

export default {
  components: {
    GlAlert,
    CodeSnippetAlert,
  },
  errorTexts: {
    [COMMIT_FAILURE]: s__('Pipelines|The GitLab CI configuration could not be updated.'),
    [DEFAULT_FAILURE]: __('Something went wrong on our end.'),
    [LOAD_FAILURE_UNKNOWN]: s__('Pipelines|The CI configuration was not loaded, please try again.'),
    [PIPELINE_FAILURE]: s__('Pipelines|There was a problem with loading the pipeline data.'),
  },
  successTexts: {
    [COMMIT_SUCCESS]: __('Your changes have been successfully committed.'),
    [DEFAULT_SUCCESS]: __('Your action succeeded.'),
  },
  props: {
    failureType: {
      type: String,
      required: false,
      default: null,
    },
    failureReasons: {
      type: Array,
      required: false,
      default: () => [],
    },
    showFailure: {
      type: Boolean,
      required: false,
      default: false,
    },
    showSuccess: {
      type: Boolean,
      required: false,
      default: false,
    },
    successType: {
      type: String,
      required: false,
      default: null,
    },
  },
  data() {
    return {
      codeSnippetCopiedFrom: '',
    };
  },
  computed: {
    failure() {
      switch (this.failureType) {
        case LOAD_FAILURE_UNKNOWN:
          return {
            text: this.$options.errorTexts[LOAD_FAILURE_UNKNOWN],
            variant: 'danger',
          };
        case COMMIT_FAILURE:
          return {
            text: this.$options.errorTexts[COMMIT_FAILURE],
            variant: 'danger',
          };
        case PIPELINE_FAILURE:
          return {
            text: this.$options.errorTexts[PIPELINE_FAILURE],
            variant: 'danger',
          };
        default:
          return {
            text: this.$options.errorTexts[DEFAULT_FAILURE],
            variant: 'danger',
          };
      }
    },
    success() {
      switch (this.successType) {
        case COMMIT_SUCCESS:
          return {
            text: this.$options.successTexts[COMMIT_SUCCESS],
            variant: 'info',
          };
        default:
          return {
            text: this.$options.successTexts[DEFAULT_SUCCESS],
            variant: 'info',
          };
      }
    },
  },
  created() {
    this.parseCodeSnippetSourceParam();
  },
  methods: {
    dismissCodeSnippetAlert() {
      this.codeSnippetCopiedFrom = '';
    },
    dismissFailure() {
      this.$emit('hide-failure');
    },
    dismissSuccess() {
      this.$emit('hide-success');
    },
    parseCodeSnippetSourceParam() {
      const [codeSnippetCopiedFrom] = getParameterValues(CODE_SNIPPET_SOURCE_URL_PARAM);
      if (codeSnippetCopiedFrom && CODE_SNIPPET_SOURCES.includes(codeSnippetCopiedFrom)) {
        this.codeSnippetCopiedFrom = codeSnippetCopiedFrom;
        window.history.replaceState(
          {},
          document.title,
          removeParams([CODE_SNIPPET_SOURCE_URL_PARAM]),
        );
      }
    },
  },
};
</script>

<template>
  <div>
    <code-snippet-alert
      v-if="codeSnippetCopiedFrom"
      :source="codeSnippetCopiedFrom"
      class="gl-mb-5"
      @dismiss="dismissCodeSnippetAlert"
    />
    <gl-alert
      v-if="showSuccess"
      :variant="success.variant"
      class="gl-mb-5"
      @dismiss="dismissSuccess"
    >
      {{ success.text }}
    </gl-alert>
    <gl-alert
      v-if="showFailure"
      :variant="failure.variant"
      class="gl-mb-5"
      @dismiss="dismissFailure"
    >
      {{ failure.text }}
      <ul v-if="failureReasons.length" class="gl-mb-0">
        <li v-for="reason in failureReasons" :key="reason">{{ reason }}</li>
      </ul>
    </gl-alert>
  </div>
</template>
