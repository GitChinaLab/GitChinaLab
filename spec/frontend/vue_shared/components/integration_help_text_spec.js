import { GlIcon, GlLink, GlSprintf } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';

import IntegrationHelpText from '~/vue_shared/components/integrations_help_text.vue';

describe('IntegrationHelpText component', () => {
  let wrapper;
  const defaultProps = {
    message: 'Click %{linkStart}Bar%{linkEnd}!',
    messageUrl: 'http://bar.com',
  };

  function createComponent(props = {}) {
    return shallowMount(IntegrationHelpText, {
      propsData: {
        ...defaultProps,
        ...props,
      },
      stubs: {
        GlSprintf,
      },
    });
  }

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  it('should use the gl components', () => {
    wrapper = createComponent();

    expect(wrapper.find(GlSprintf).exists()).toBe(true);
    expect(wrapper.find(GlIcon).exists()).toBe(true);
    expect(wrapper.find(GlLink).exists()).toBe(true);
  });

  it('should render the help text', () => {
    wrapper = createComponent();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not use the gl-link and gl-icon components', () => {
    wrapper = createComponent({ message: 'Click nowhere!' });

    expect(wrapper.find(GlSprintf).exists()).toBe(true);
    expect(wrapper.find(GlIcon).exists()).toBe(false);
    expect(wrapper.find(GlLink).exists()).toBe(false);
  });

  it('should not render the link when start and end is not provided', () => {
    wrapper = createComponent({ message: 'Click nowhere!' });

    expect(wrapper.element).toMatchSnapshot();
  });
});
