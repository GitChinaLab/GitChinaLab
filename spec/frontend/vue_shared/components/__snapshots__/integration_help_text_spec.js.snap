// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntegrationHelpText component should not render the link when start and end is not provided 1`] = `
<span>
  Click nowhere!
</span>
`;

exports[`IntegrationHelpText component should render the help text 1`] = `
<span>
  Click 
  <gl-link-stub
    href="http://bar.com"
    target="_blank"
  >
    
        Bar
        
    <gl-icon-stub
      class="gl-vertical-align-middle"
      name="external-link"
      size="12"
    />
  </gl-link-stub>
  !
</span>
`;
